src/main/java/ecdar/backend/EngineConnection.java

```diff
@@ -39,6 +39,10 @@ public Engine getEngine() {
         return engine;
     }
 
+    public int getPort() {
+        return Integer.parseInt(getStub().getChannel().authority().split(":", 2)[1]);
```

Comment:
A member variable with a name such as ``final int port`` should be added to this class (To act as a read-only backing field for this getter). Then upon construction, we should initialize it to the value produced by this expression.
````
private final int port;

EngineConnection(Engine engine, Process process, EcdarBackendGrpc.EcdarBackendStub stub, ManagedChannel channel)
    ...
    port = Integer.parseInt(getStub().getChannel().authority().split(":", 2)[1]
}

public int getPort() {
    return port;
}
````

Reply:
That is a good suggestion, I will push this as a change